lumigator/infra/mzai/helm/lumigator/templates/gen_secrets.yaml

```diff
@@ -0,0 +1,13 @@
+{{- include "lumigator.consts" . -}}
+{{- if not .Values.mistralSecretName -}}
+{{- if not .Values.mistralKey -}}
+{{- fail "Neither an existing Mistral API Secret name nor Mistral API key details have been provided" -}}
```

Comment:
For me, it's not an error to not include a secret with the key or the key instead. Maybe some users don't have a Mistral API key because they don't use it

Reply:
Hmmmm, good point. In that case, should we add the MISTRAL_API_KEY to the pod environment, or should we skip it?

Reply:
The actual behavior is not include it. But I'm not 100% sure that we expect that in the code. [Here](https://github.com/mozilla-ai/lumigator/blob/main/lumigator/python/mzai/backend/services/completions.py#L19) is where it's used, and we rely in the library to rise the error (if I'm not wrong). Maybe @veekaybee can help us with this 😄

Reply:
We expect the Mistral completion service to work when the user is interested in performing a ground truth evaluation using the Mistral API, that is it's entirely open to the user if they want to use the component or not. 

Right now this passed at the "SDK", aka [notebook level](https://github.com/mozilla-ai/lumigator/blob/4ff0acffecbcdd5a9394e3097f1a6ed2c45bd4c5/notebooks/lumigator_demo.py#L342). So not having the key will not stop the app from working, and if you want to perform an eval, you will need to inject a key at eval time. How we implement this depends more on the SDK side.  Generally in the LLM space, the expected user behavior is for the user to provide the key in the application they're developing by injecting the token. As an example: 

https://github.com/run-llama/llama_index?tab=readme-ov-file#-example-usage

Reply:
Ok, I can try to include some way of signalling that the Mistral API key is not used (post install hook?) so the user is aware that this is not available.

Reply:
For me it's enough with what you did @javiermtorres, great work!